pkg/kube/sourcerepositories.go

```diff
@@ -56,70 +86,57 @@ func GetOrCreateSourceRepositoryCallback(jxClient versioned.Interface, ns string
 	resourceName := naming.ToValidName(organisation + "-" + name)
 
 	repositories := jxClient.JenkinsV1().SourceRepositories(ns)
-	description := fmt.Sprintf("Imported application for %s/%s", organisation, name)
 
 	providerName := ToProviderName(providerURL)
 
-	labels := map[string]string{}
-	sr := &v1.SourceRepository{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "SourceRepository",
-			APIVersion: jenkinsio.GroupName + "/" + jenkinsio.Version,
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   resourceName,
-			Labels: labels,
-		},
-		Spec: v1.SourceRepositorySpec{
-			Description:  description,
-			Org:          organisation,
-			Provider:     providerURL,
-			ProviderName: providerName,
-			Repo:         name,
-		},
+	foundSr, err := FindSourceRepository(jxClient, ns, organisation, name, providerName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to find existing SourceRepository")
 	}
+
+	// If we did not find an existing SourceRepository for this org/repo, create one
```

Comment:
It beats me why we would use a callback here? any ideas?

Reply:
Basically, in most cases, the caller just needs a fairly basic set to be set in the SR, but when import’s `addProwConfig` calls it at https://github.com/jenkins-x/jx/blob/d3105b309c55d150328bfe8dd7f7f0081f566d1c/pkg/cmd/importcmd/import.go#L1036, it needs to set more information that isn’t necessarily available yet in other cases.
 
I thought about changing other callers to just lookup, since at this point, the SR should actually always already exist, but because of manually specified SRs in the dev env repo, we can’t be 100% sure those SRs are properly labeled yet, so those fields are ones we do need to ensure are there and add if they’re not.

Reply:
you would not need a callback for that, right? anyways